android/rctmgl/src/main/java/com/mapbox/rctmgl/components/mapview/RCTMGLMapView.java

```diff
@@ -171,6 +172,20 @@ public RCTMGLMapView(Context context, RCTMGLMapViewManager manager, MapboxMapOpt
         setLifecycleListeners();
     }
 
+    @Override
```

Comment:
Instead of listening for window attachment changes, lets hook into the actual visibility of the window. This works for both backgrounding, and killing an activity (onStop vs onDestroy)